R/utils_get_db.R

```diff
@@ -217,11 +217,58 @@ get_pkg_info <- function(pkg_name, db_name = golem::get_golem_options('assessmen
 #' @noRd
 get_metric_weights <- function(db_name = golem::get_golem_options('assessment_db_name')){
   dbSelect(
-    "SELECT name, weight
+    "SELECT name, weight, is_perc
      FROM metric", db_name
   )
 }
 
+#' Get Assessments
+#'
+#' Retrieves metric name and current weight from metric table
+#' 
+#' @param pkg_name character name of the package
+#' @param db_name character name (and file path) of the database
+#'
+#' @returns a data frame
+#' @noRd
+get_assess_blob<- function(pkg_name, db_name = golem::get_golem_options('assessment_db_name')) {
+  db_table <- dbSelect("SELECT metric.name, package_metrics.encode FROM package 
+                       INNER JOIN package_metrics ON package.id = package_metrics.package_id
+                       INNER JOIN metric ON package_metrics.metric_id = metric.id
+                       WHERE package.name = {pkg_name}", 
+                       db_name = db_name)
+  
+  purrr::pmap_dfc(db_table, function(name, encode) {dplyr::tibble(unserialize(encode)) %>% purrr::set_names(name)}) 
+}
+
+
+#' Get Dependency Pkg Versions and Scores
+#'
+#' 
+#' @param pkg_name character name of the package
+#' @param verify_data a data.frame used to verify whether a pkg exists in the db
+#'
+#' @returns a list
+#' @noRd
+get_versnScore <- function(pkg_name, verify_data) {
+  
+  riskmetric_assess <- 
+    riskmetric::pkg_ref(pkg_name,
+                        source = "pkg_cran_remote",
+                        repos = c("https://cran.rstudio.com")) %>%
+    dplyr::as_tibble() 
+  
+  
+  if (pkg_name %in% verify_data$name) { #loaded2_db()$name
+    pkg_score <- verify_data %>% filter(name == pkg_name) %>% pull(score) %>% as.character
+  } else {
+    pkg_score <- ""
+  } 
+  
+  return(list(name = riskmetric_assess$package, version = riskmetric_assess$version, score = pkg_score))   
+}
```

Comment:
@Robert-Krajcik, are you just calling `pkg_ref` here to get the latest version for each package? I don't think that's a good strategy since the package could have been uploaded 10 months ago, which means the version you're listing here wouldn't be applicable. I don't think we should even report any pkg versions on this tab since we don't know what the pkg authors are using. In fact, could we also lose the "Name" field here? It's redundant of "Package", which we should keep because it contains the bounds on version numbers, if they exist. If you need to keep "Name" for the upload button to add new pkgs, I think you can create "invisible" columns in `DT` output.

![image](https://github.com/pharmaR/riskassessment/assets/17878953/c031565e-8984-4201-a5c8-2a433110fc7b)

Reply:
Yes I need the name to upload new packages. But I can make it invisible.

Reply:
`Also, I don't think the desired behavior is to redirect to see cli dependencies after upload. I think it would be more appropriate to leave the control panel on gt and just update the score and review package button to be an arrow. Thoughts?`

My thinking was that pressing the upload button should behave the same as though the user had pressed the right arrow button.  The only difference is that there is this intermediate step of loading the package.